mantidproject/mantid
# SANS Call Convert To Wavelength Once

**Description of work.**
In the existing code we currently call ConvertToWavelength on the full dataset, extract the selected wavelength rage, then discard our full wavelength data to repeat the entire process.
This changes the underlying algorithm to accept a list of wavelength pairs so it can internally loop over each selected wavelength range. This can reduce the runtime by >50% depending on the number of wavelength pairs specified

**To test:**
- Download the following test data / mask files ...etc: 
[wav_loops.zip](https://github.com/mantidproject/mantid/files/7473995/wav_loops.zip)
- Extract this to a known dir
- Start Mantid, go to Manage User Directories and add the unzipped folder
- Go to ISIS SANS, click user file and open the only `.toml` file in the unzipped folder
- Click Batch file and select the only CSV in that folder
- Go to settings (left hand side), then `Q, Wavelength....` and under the `Wavelength Ranges` box ensure a long comma separated string is present
- Go back to runs (LHS), the in the bottom right ensure memory is ticked. Click process all and ensure data is produced `66946_rear_1D` for 1D and 2D mode (re-run after both). 
- The data itself will look garbage, as the test data is simply two empties combined and the system tests should check for data consistency

Fixes #31866



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
